### DISCUSSION TITLE
I cant figure out the filters. Here's what I've tried.

### DISCUSSION THREAD

**Discussion by laptop-doug** [WinMerge/winmerge#2325] [General]
EDIT:  oh hell, do i have to specify what filters, when i pick the dirs to compare?

i have this folder, in one of the compared dirs.
`C:\Users\asdf\Documents\asdf\d_theme_3\inc`
I have a few of these folders, scattered around
`_gsdata_`

here's what i've tried:
 
```
def: include
d: \\_gsdata_$
 
d: \\_gsdata_\$
 
d: \\\_gsdata_$ 

d: \\\_gsdata_\$ 

d: \_gsdata_$ 

d: \\\_gsdata_$ 

d: \_gsdata_\$ 

d: _gsdata_$  
```
and similar iterations for the specific dir.  
If I change the filter, in notepad (in the C:\Program Files\WinMerge\Filters dir) and then refresh, does that work?
If I change the filter, from the Filters dialog, then refresh, does that work?
Do I have to delete and reinstall the filter?

**Comment by sdottaka** [comment]
`d: \\_gsdata_$` should work.

> If I change the filter, in notepad (in the C:\Program Files\WinMerge\Filters dir) and then refresh, does that work?

If you edit with Notepad launched from a location other than the Filter dialog, you will need to restart WinMerge for the changes to take effect.

> If I change the filter, from the Filters dialog, then refresh, does that work?

That should be the case as long as you are using a recent version of WinMerge. However, if your filters are stored in C:\Program Files\WinMerge\Filters, you may not have write permissions unless you start WinMerge with administrator privileges.

> Do I have to delete and reinstall the filter?

I don't think that's necessary.

![image](https://github.com/WinMerge/winmerge/assets/98126/ee631f4f-ee11-4390-aaf3-21f41f2471e3)

**Comment by laptop-doug** [comment]
ok, here's how it works.  When you select your folders for compare, you choose 1 (one) filter.  Only that filter is active now.  You can change the filter, outside of Winmerge, open in notepad, from windows.  When you make a change, you hit F5 to refresh and your changes are reflected.
![image](https://github.com/WinMerge/winmerge/assets/105393749/3377cc04-9a73-4a39-8453-0bd21924eb8c)

Now, the docs, I finally found "Applying File Filters"  But it says there is a 'Filter Field', but there isnt.
_In the Filters dialog, use the Filefilters tab to choose a file filter and load it in the Filter field.
Note If a file filter is currently enabled, the selected file filter is already loaded in the Filter field._
![image](https://github.com/WinMerge/winmerge/assets/105393749/f5ebde3a-fda3-4b3e-8cf3-0898507e76b3)

**Reply by sdottaka** [reply]
Oh, I didn't think that the filter changes made in the external editor would be reflected immediately by pressing the F5 key.
